app/controllers/platform/geolocation.js

```diff
@@ -0,0 +1,107 @@
+var log = require("log");
+
+/**
+ * The scoped constructor of the controller.
+ **/
+(function constructor() {
+    var Map = require('ti.map');
+
+    Titanium.Geolocation.preferredProvider = Titanium.Geolocation.PROVIDER_GPS;
+    Titanium.Geolocation.accuracy = Titanium.Geolocation.ACCURACY_BEST;
+    Titanium.Geolocation.distanceFilter = 10;
+
+    //Check for Google Play Services. In order to view maps, Google Play services needs to be installed on the device
+    if (OS_ANDROID) {
+        var rc = Map.isGooglePlayServicesAvailable();
+        switch (rc) {
+        case Map.SUCCESS:
+            Ti.API.info('Google Play services is installed.');
+            break;
+        case Map.SERVICE_MISSING:
+            alert('Google Play services is missing. Please install Google Play services from the Google Play store.');
+            break;
+        case Map.SERVICE_VERSION_UPDATE_REQUIRED:
+            alert('Google Play services is out of date. Please update Google Play services.');
+            break;
+        case Map.SERVICE_DISABLED:
+            alert('Google Play services is disabled. Please enable Google Play services.');
+            break;
+        case Map.SERVICE_INVALID:
+            alert('Google Play services cannot be authenticated. Reinstall Google Play services.');
+            break;
+        default:
+            alert('Unknown error.');
+        }
+    }
+
+    //checks for location service available
+    if (Titanium.Geolocation.locationServicesEnabled) {
+
+        //adds listener to app and pops alert when location found
+        Ti.App.addEventListener("app:got.location", function(d) {
```

Comment:
Please try to avoid global event listeners by using callbacks across your scope instead. Let me know if I can help with that.